### PR TITLE
added getCollectionSize method for mysql mariadb postgres and mongodb…

### PR DESCRIPTION
**Pull request by faisalill**

### CODE REVIEW

**Comment by faisalill** [comment]
It works for MySQL, MariaDB and Postgres (I tested it). I don't know about mongodb becuase I tried to use the UTOPIA\MONGO\CLIENT and it is just crashing the server. So I used the MongoDB\Client and made a function to get the size in bytes and it worked so I have implemented a similar thing for the utopia mongo client class.

File: src/Database/Adapter/MariaDB.php [hunk 1]
```diff
@@ -141,6 +141,43 @@ public function createCollection(string $name, array $attributes = [], array $in
...
+         $query->bindParam(':name', $name);
+         $query->execute();
+         try {
+            $query->execute();
+            $size = $query->fetchColumn();
+        } catch (PDOException $e) {
+            throw new Exception("Invalid table name");
+        }
+     
+         return (int) $size;
```
Comment (fogelito) [inline src/Database/Adapter/MariaDB.php]: Not sure we need the casting? need to be tasted with big numbers..

File: src/Database/Adapter/Postgres.php [hunk 2]
```diff
@@ -147,6 +147,35 @@ public function createCollection(string $name, array $attributes = [], array $in
...
+     * 
+     */
+
+     public function getCollectionSize(string $name): int
+     {
+         $database = $this->getDefaultDatabase();
+         $name = $this->filter($name);
+     
+         $query = $this->getPDO()->prepare("
+         SELECT pg_size_pretty(pg_total_relation_size(
```
Comment (fogelito) [inline src/Database/Adapter/Postgres.php]: Please double-check with pg_size_pretty() as we want data in bytes?

File: src/Database/Adapter/MariaDB.php [hunk 3]
```diff
@@ -141,6 +141,43 @@ public function createCollection(string $name, array $attributes = [], array $in
...
 
+    /**
+     * Get Collection Size
+     * @param string $name
+     * @return int
+     * @throws Exception
+     * 
+     */
+
+     public function getCollectionSize(string $name): int
```
Comment (fogelito) [inline src/Database/Adapter/MariaDB.php]: Maybe we need to a datatype bigger than int?

**Comment by faisalill** [comment]
Thanks for the review. I will work on it

File: src/Database/Adapter/MariaDB.php (hunk 1 above)
Comment (faisalill) [reply src/Database/Adapter/MariaDB.php]: changed it to a string.

File: src/Database/Adapter/Postgres.php (hunk 2 above)
Comment (faisalill) [reply src/Database/Adapter/Postgres.php]: I removed getCollectionSize for Postgres as it works but is giving erros in the tests.

File: src/Database/Adapter/MariaDB.php (hunk 3 above)
Comment (faisalill) [reply src/Database/Adapter/MariaDB.php]: I changed the return type to string.

**Pull request merged**